### PR TITLE
Added heartbeat to API

### PR DESCRIPTION
**Pull request by toolunious**
Related issue: https://github.com/Amsterdam/GOB-API/issues/91

*Problem*: API service has no heartbeat at the moment

*Solution*: 
 * Added service orchestration in API application configurable with env `API_INFRA_SERVICES` (comma separated list of service references as defined in `services:registry`).
 * Added function to run services as thread.
 * Added a service implementation that starts/stops a `gobcore.message_broker.messagedriven_service`.
 * Added "start all" routines to both uwsgi and flask.run code paths.

*POW*:
image
*`gob-management/status` API service recognised by heartbeat*

### CODE REVIEW

File: src/gobapi/config.py [hunk 1]
```diff
@@ -15,3 +15,8 @@
     'host': os.getenv("DATABASE_HOST_OVERRIDE", "localhost"),
     'port': os.getenv("DATABASE_PORT_OVERRIDE", 5406),
 }
+
+# see gobapi.services.registry
+API_INFRA_SERVICES = os.getenv(
+    "API_INFRA_SERVICES", ""
+).upper().split(",")
```
Comment (Vegter) [inline src/gobapi/config.py]: Question: Should any value be set when running API outside docker?

File: src/gobapi/services.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    """ Start a threaded service """
+    service_thread = threading_backend(
+        target=service.start, name=service.name
+    )
+
+    def _teardown_func():
+        logger.info(f"Stopping service {service.name}")
+        service.stop()
+        service_thread.join(4)
+
```
Comment (Vegter) [inline src/gobapi/services.py]: Can you comment on 4 or give it a descriptive name?

File: src/gobapi/services.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+class Service(abc.ABC):
+    name = "Service"
+    backend = None
+
+    def __init__(self, backend=None):
+        if backend:
+            self.backend = backend
+
```
[the remark this answers is not in the document]
Comment (toolunious) [reply src/gobapi/services.py]: f0e6f370686b146399a825abe56c614d157c89c5

File: src/gobapi/config.py (hunk 1 above)
Comment (toolunious) [reply src/gobapi/config.py]: 71b575010b042d34f07b61faad313bf8b48570c2

File: src/gobapi/services.py (hunk 2 above)
Comment (toolunious) [reply src/gobapi/services.py]: 96ed1b5f4270c7ac81b138247cfb496cb2abe1a9

File: src/gobapi/infra.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ThreadFactory = typing.Callable[[services.Service], threading.Thread]
+
+
+def start_all_services(
+            service_idents: list,
+            registry: dict = services.registry,
+            factory: ThreadFactory = services.threaded_service
+        ) -> typing.List[threading.Thread]:
+    threads = []
+    for service_ident in service_idents:
```
[the remark this answers is not in the document]
Comment (toolunious) [reply src/gobapi/infra.py]: 203fc3f717bee2bd89ca543e9b7acc281b480d5c

**Pull request merged**